src/operations/resolve.ts

```diff
@@ -0,0 +1,159 @@
+import { Store } from '../store';
+import {
+  Data,
+  SelectionSet,
+  OperationRequest,
+  Variables,
+  Fragments,
+  DataField,
+} from '../types';
+import {
+  getSelectionSet,
+  getMainOperation,
+  normalizeVariables,
+  getFragments,
+  getName,
+  getFieldArguments,
+  getFieldAlias,
+} from '../ast';
+import { SelectionIterator } from './shared';
+import { joinKeys, keyOfField, isDataOrKey } from '../helpers';
+import warning from 'warning';
+
+interface Context {
+  store: Store;
+  variables: Variables;
+  fragments: Fragments;
+}
+
+export const resolveData = (
+  store: Store,
+  request: OperationRequest,
+  data: Data
+) => {
+  const operation = getMainOperation(request.query);
```

Comment:
This is not wrapped in the initialiser / clearer for store state. We still need the new dependencies for queries 👍